core/templates/services/playthrough-issues-backend-api.service.ts

```diff
@@ -104,10 +104,14 @@ export class PlaythroughIssuesBackendApiService {
         exp_issue_dict: issueToResolve.toBackendDict(),
         exp_version: expVersion
       }).toPromise().then(() => {
-      let issueIndex = this.cachedIssues !== null ?
-        this.cachedIssues.findIndex(issue => {
+      let issueIndex;
+      if (this.cachedIssues !== null) {
+        issueIndex = this.cachedIssues.findIndex(issue => {
           return angular.equals(issue, issueToResolve);
-        }) : -1;
+        });
+      } else {
+        issueIndex = -1;
+      }
       if (issueIndex === -1) {
```

Comment:
OK, this code is much clearer to me now, thanks! That said, I think it can be made even more clear - please change the `then` block to:

```typescript
if (this.cachedIssues !== null) {
  let issueIndex = this.cachedIssues.findIndex(
    issue => angular.equals(issue, issueToResolve));
  if (issueIndex !== -1) {
    this.cachedIssues.splice(issueIndex, 1);
    return;
  }
}
throw Error(
  'An issue which was not fetched from the backend has been ' +
  'resolved'));
```

Reply:
This is throwing frontend_tests errors, from spec.ts file of spy success and failure.

Reply:
This one as well, please add me to that PR.